source/expectations/data_values/ut_compound_data_value.tpb

```diff
@@ -194,105 +194,52 @@ create or replace type body ut_compound_data_value as
-                       ) src
-                       on (tgt.item_no = src.item_no and tgt.data_id = src.data_id)
-                       when matched then update
-                       set tgt.item_hash = src.item_hash,
-                           tgt.pk_hash = src.pk_hash,
-                           tgt.duplicate_no = src.duplicate_no]'
-                       using a_exclude_xpath, a_include_xpath,a_join_by_xpath,self.data_id, l_other.data_id;
+    **/   
+    ut_compound_data_helper.update_row_and_pk_hash(self.data_id, l_other.data_id, a_exclude_xpath,a_include_xpath,a_join_by_xpath);
     
-    /* Peform minus on two sets two get diffrences that will be used later on to print results */
-    execute immediate 'insert into ' || l_ut_owner || '.ut_compound_data_diff_tmp ( diff_id,item_hash,pk_hash,duplicate_no)
-                       with source_data as
-                       ( select t.data_id,t.item_hash,t.duplicate_no,
-                           pk_hash
-                           from  ' || l_ut_owner || '.ut_compound_data_tmp t
-                           where data_id = :self_guid or data_id = :other_guid
-                        )           
-                       select distinct :diff_id,tmp.item_hash,tmp.pk_hash,tmp.duplicate_no
-                       from(
-                         (
-                           select t.item_hash,t.duplicate_no,t.pk_hash
-                           from  source_data t
-                           where t.data_id = :self_guid
-                           minus
-                           select t.item_hash,t.duplicate_no,t.pk_hash
-                           from  source_data t
-                           where t.data_id = :other_guid
-                         )
-                           union all
-                         (
-                           select t.item_hash,t.duplicate_no,t.pk_hash
-                           from  source_data t
-                           where t.data_id = :other_guid
-                           minus
-                           select t.item_hash,t.duplicate_no,t.pk_hash
-                           from  source_data t
-                           where t.data_id = :self_guid
-                        ))tmp'
-       using self.data_id, l_other.data_id,
-             l_diff_id, 
-             self.data_id, l_other.data_id,
-             l_other.data_id,self.data_id;
-   --result is OK only if both are same
-    if sql%rowcount = 0 and self.elements_count = l_other.elements_count then
+    /*!* 
+    * Comparision is based on type of search, for inclusion based search we will look for left join only.
+    * For normal two side diff we will peform minus on two sets two get diffrences.
+    * SELF is expected. 
+    * Due to growing complexity I have moved a dynamic SQL into helper package.
+    */
+    l_sql := ut_compound_data_helper.get_refcursor_matcher_sql(l_ut_owner,a_inclusion_compare,a_is_negated);
+
+    execute immediate l_sql
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a WHEN OTHERS exception handler to ensure that all exceptions are trapped. ([plsql:ExecuteImmediateTrapExceptionsCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AExecuteImmediateTrapExceptionsCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=774&issues=AWbTzUC8O08i52vd_XcV&open=AWbTzUC8O08i52vd_XcV)

[ISSUE_KEY]: AWbTzUC8O08i52vd_XcV